enterprise/suppress-user/repo_switch.go

```diff
@@ -0,0 +1,56 @@
+package suppression
+
+import (
+	"io"
+	"sync"
+
+	"github.com/rudderlabs/rudder-server/enterprise/suppress-user/model"
+)
+
+// RepoSwitcher is a repository that can be be used to switch repository at runtime
```

Comment:
Since this repo switcher lives in the `suppress-user` package I think it would make sense to explain a bit why it's being used. What you wrote in the PR description would be a good enough explanatation to help our future-selves:

Therefore I'd say that some sort of of version of the following would be a nice addition:

> Here, we make sure that gateway starts once we have the latest users suppressions. Also, we start getting the full suppression list asynchronously in a separate badgerdb repo, since it might take sometime. Once, we have the full list in badgerdb. We simply swap the old badgerdb (with latest users) with the new badgerdb (with all users).

Alternatively we can add this comment in `factory.go` where this is used.